python/cudf/cudf/core/buffer/spillable_buffer.py

```diff
@@ -465,148 +360,114 @@ def memoryview(
-        return SpillableBufferSlice(
-            base=self._base, offset=offset + self._offset, size=size
-        )
+    _owner: SpillableBufferOwner
 
-    @classmethod
-    def deserialize(cls, header: dict, frames: list):
-        # TODO: because of the hack in `SpillableBuffer.serialize()` where
-        # frames are of type `Buffer`, we always deserialize as if they are
-        # `SpillableBuffer`. In the future, we should be able to
-        # deserialize into `SpillableBufferSlice` when the frames hasn't been
-        # copied.
-        return SpillableBuffer.deserialize(header, frames)
-
-    def memoryview(
-        self, *, offset: int = 0, size: Optional[int] = None
-    ) -> memoryview:
-        size = self._size if size is None else size
-        return self._base.memoryview(offset=self._offset + offset, size=size)
-
-    def __repr__(self) -> str:
-        return (
-            f"<SpillableBufferSlice size={format_bytes(self._size)} "
-            f"offset={format_bytes(self._offset)} of {self._base} "
-        )
-
-    # The rest of the methods delegate to the base buffer.
     def spill(self, target: str = "cpu") -> None:
-        return self._base.spill(target=target)
+        return self._owner.spill(target=target)
 
     @property
     def is_spilled(self) -> bool:
-        return self._base.is_spilled
+        return self._owner.is_spilled
 
     @property
     def exposed(self) -> bool:
-        return self._base.exposed
+        return self._owner.exposed
 
     @property
     def spillable(self) -> bool:
-        return self._base.spillable
+        return self._owner.spillable
 
     def spill_lock(self, spill_lock: SpillLock) -> None:
-        self._base.spill_lock(spill_lock=spill_lock)
+        self._owner.spill_lock(spill_lock=spill_lock)
 
     def memory_info(self) -> Tuple[int, int, str]:
-        (ptr, _, device_type) = self._base.memory_info()
+        (ptr, _, device_type) = self._owner.memory_info()
         return (ptr + self._offset, self.nbytes, device_type)
+
+    def mark_exposed(self) -> None:
+        self._owner.mark_exposed()
+
+    def serialize(self) -> Tuple[dict, list]:
+        """Serialize the Buffer
+
+        Normally, we would use `[self]` as the frames. This would work but
+        also mean that `self` becomes exposed permanently if the frames are
+        later accessed through `__cuda_array_interface__`, which is exactly
+        what libraries like Dask+UCX would do when communicating!
+
+        The sound solution is to modify Dask et al. so that they access the
+        frames through `.get_ptr()` and holds on to the `spill_lock` until
+        the frame has been transferred. However, until this adaptation we
+        use a hack where the frame is a `Buffer` with a `spill_lock` as the
+        owner, which makes `self` unspillable while the frame is alive but
+        doesn't expose `self` when `__cuda_array_interface__` is accessed.
```

Comment:
We should reassess this evaluation after the next PR unifying COW and spilling. I don't think I agree with this statement, it implies leaking knowledge of cudf buffer internals to dask. Once we've finished the unification we should revisit whether there's a more API-friendly way of doing this. If not, we need to think about the appropriate generalization of our exposure semantics to generic CAI usage.

Reply:
FWIW, this approach (as the code in this PR uses) is (morally) the way pytorch ensures lifetime of a `CAI`-supporting object being turned into a torch tensor